### ISSUE TITLE
Remove font-size inheritance on paragraph to heading transform

### ISSUE THREAD

**Issue by Humanify-nl** [WordPress/gutenberg#39347]
### Description

Font-size set on a paragraph that is later transformed to a heading is inherited with the transform. 

After some searching I found this issue [34590](https://github.com/WordPress/gutenberg/issues/34590) about inheriting font-sizes. And the global idea to inherit styles (if supported) is a great idea. But the following use-case is breaking when theme.json control is involved:

Theme author sets in theme.json the following in block-settings:

	"core/heading": {
		"typography": {
			"fontSizes": []
		}
	}

Now, when the large size paragraph transformed with the heading has inherited a large font size, this **can not be remove**d in the editor.

Because there are no controls available (also pressing 'reset all' in the typography section in inspector is not removing the style.)

**The thinking behind this?**
I think the idea behind this, that in an ideal scenario the font-sizes for headings & paragraph are unified, so you set 6 global font-sizes that apply for headings and paragraphs.

But in many use cases, we want to set different sizes for paragraphs (normal, large, small) and not intertwine these values with the sizes for headings. 

### Step-by-step reproduction instructions

1. Disallow font-size controls for heading inside theme.json block settings (see above for theme.json fragment). 
2. Set a paragraph font-size to large in editor.
3. Transform to heading (which now inherits the large font size).
4. Try to change the heading size back to what it should be.

### Environment info

- WP 5.9.1
- Gutenberg 12.7.0

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by paaljoachim** [comment]
@oandregal might know something about this issue.

**Comment by Humanify-nl** [comment]
Thanks! 

I encountered this again in a more typical use case. 

When you add a cover block and start typing in the paragraph and set it to large, then transform it to a heading (which is natively larger), the heading will inherit the sizes (is-medium-fonts-size) of paragraph. 

Inheriting styles with transforms is, I believe, too complex to handle with the many use cases that can break it.

**Comment by paaljoachim** [comment]
This does sound kinda like a bug:

> When you add a cover block and start typing in the paragraph and set it to large, then transform it to a heading (which is natively larger), the heading will inherit the sizes (is-medium-fonts-size) of paragraph.

**Comment by Humanify-nl** [comment]
> This does sound kinda like a bug:
> 
> > When you add a cover block and start typing in the paragraph and set it to large, then transform it to a heading (which is natively larger), the heading will inherit the sizes (is-medium-fonts-size) of paragraph.

Yes, I think this is more prevalent because placing a cover-block auto-assigns font-size large to the text in my case (wp 5.9.2). 

Should I make a specific bug report for this?

**Comment by paaljoachim** [comment]
Hey @Humanify-nl Robbert

I went ahead and added the "Bug" label and "Needs Testing" label, as we need some additional folks to test it out.

**Comment by ndiego** [comment]
This issue was reviewed in today's Editor Bug Scrub. We have been able to replicate using the latest versions of Gutenberg and WordPress. Therefore, I am going to remove the "Needs Testing" label and add the "Needs Technical Feedback". While this might be considered a bug, fixing this has a large implication on block transforms and inherited styles.

**Comment by youknowriad** [comment]
Re-tested on trunk: transforming a paragraph with a font-size preset into a heading no longer carries the font size over, so themes that disable heading sizes aren't stuck with an unremovable value. Closing as fixed.
